MekHQ/src/mekhq/gui/baseComponents/JDisableablePanel.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2021 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MekHQ.
+ *
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ. If not, see <http://www.gnu.org/licenses/>.
+ */
+package mekhq.gui.baseComponents;
+
+import javax.swing.*;
+import java.awt.*;
+
+/**
+ * This panel is used when you want a JPanel that enables/disables all of its children when it is
+ * enabled/disabled. This is purposefully not recursive, as we only want child components to be
+ * disabled based on the settings here.
+ */
+public class JDisableablePanel extends JPanel {
+    //region Constructors
+    public JDisableablePanel(final String name) {
+        super();
+        setName(name);
+    }
+    //endregion Constructors
+
+    /**
+     * This override forces all child components to be the same value for enabled as this component,
+     * thus allowing one to easily enable/disable child components and panels.
+     * @param enabled whether to enable the child components or not
+     */
+    @Override
+    public void setEnabled(final boolean enabled) {

```

Comment:
I feel that for something like this I would do some type of generic function that takes a java.awt.Container, and walks the list of getComponents() enabling/disabling them (and all their subcomponents). That would work for anything that inherits from that class, so you could disable other UI elements like Tabs or Trees.

It could be a recursive call, or just use a queue, and once you dequeue something to change their setEnable, you enqueue their sub-components.

Reply:
I want this to only disable the components, not their subcomponents. If we want to disable the subcomponents, we just make that extend this panel (or another base component, but I cannot think of a good reason to have this being part of another base component currently)

Reply:
I would probably then do something like setEnabled(component, state, recurse), instead of having JDisablePanel, JDisableTab... But at the same time, if you think only JPanel is going to need this, then that's it.